PlusRemote/Logic/vtkSlicerPlusRemoteLogic.cxx

```diff
@@ -0,0 +1,580 @@
+  this->CmdGetReconstructorDeviceIDs->SetCommandName("RequestDeviceIds");
+  this->CmdGetReconstructorDeviceIDs->SetCommandAttribute("DeviceType", "VirtualVolumeReconstructor");
+  this->CmdGetReconstructorDeviceIDs->SetBlocking(this->CommandBlocking);
+  this->CmdGetReconstructorDeviceIDs->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdStartVolumeReconstruction->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdStartVolumeReconstruction->SetCommandName("StartVolumeReconstruction");
+  this->CmdStartVolumeReconstruction->SetBlocking(this->CommandBlocking);
+  this->CmdStartVolumeReconstruction->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdStopVolumeReconstruction->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdStopVolumeReconstruction->SetCommandName("StopVolumeReconstruction");
+  this->CmdStopVolumeReconstruction->SetBlocking(this->CommandBlocking);
+  this->CmdStopVolumeReconstruction->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdReconstructRecorded->SetCommandTimeoutSec(this->CommandTimeoutSec * 2.0);  // Potentially requires more time. Expiry is doubled
+  this->CmdReconstructRecorded->SetCommandName("ReconstructVolume");
+  this->CmdReconstructRecorded->SetBlocking(this->CommandBlocking);
+  this->CmdReconstructRecorded->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdStartRecording->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdStartRecording->SetCommandName("StartRecording");
+  this->CmdStartRecording->SetBlocking(this->CommandBlocking);
+  this->CmdStartRecording->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdStopRecording->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdStopRecording->SetCommandName("StopRecording");
+  this->CmdStopRecording->SetBlocking(this->CommandBlocking);
+  this->CmdStopRecording->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdGetVolumeReconstructionSnapshot->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdGetVolumeReconstructionSnapshot->SetCommandName("GetVolumeReconstructionSnapshot");
+  this->CmdGetVolumeReconstructionSnapshot->SetBlocking(this->CommandBlocking);
+  this->CmdGetVolumeReconstructionSnapshot->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdUpdateTransform->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdUpdateTransform->SetCommandName("UpdateTransform");
+  this->CmdUpdateTransform->SetBlocking(this->CommandBlocking);
+  this->CmdUpdateTransform->SetDeviceID(this->CommandIGTLIODeviceID);
+
+  this->CmdSaveConfig->SetCommandTimeoutSec(this->CommandTimeoutSec);
+  this->CmdSaveConfig->SetCommandName("SaveConfig");
+  this->CmdSaveConfig->SetBlocking(this->CommandBlocking);
+  this->CmdSaveConfig->SetDeviceID(this->CommandIGTLIODeviceID);
+
+}
+
+//----------------------------------------------------------------------------
+vtkSlicerPlusRemoteLogic::~vtkSlicerPlusRemoteLogic()
+{
+  this->CmdGetCaptureDeviceIDs->RemoveAllObservers();
```

Comment:
Is this necessary? Doesn't deleting the VTK object deletes all the observers? If not, then should we remove all observers in the destructor?